### DISCUSSION TITLE
Reduce Dependencies

### DISCUSSION THREAD

**Discussion by abhibeckert** [getsentry/sentry-php#1515] [Q&A]
### Problem Statement

I'd really like it if this package had fewer dependencies - composer installed 21 packages when I installed Sentry just now (it has more than 21 dependencies, but those are the ones I didn't already have).

It took me a couple of hours to figure out what changes I needed to make to my composer file to be able to install Sentry at all. Some of your dependencies had dependencies with dependencies that didn't match the version numbers required by dependencies of some of my dependencies... next I need to work out if the changes I made are going to cause problems elsewhere...

### Solution Brainstorm

I'd feel a lot more comfortable using Sentry if it had less dependencies. Preferably none. I feel like this is achievable considering what this package does is relatively simple.

**Comment by Jean85** [comment]
First and foremost: 
> considering what this package does is relatively simple.

I'm sorry, but this is a big understatement. Please read through all the code that it's in here before expressing something like this. This packages is now so slim because it went through a lot of work (disclaimer: mine too) and also exactly because it leverages those dependencies.

----

Ignoring this, and going over the current dependencies, these are the current ones:

https://github.com/getsentry/sentry-php/blob/9e53a01d0ee524c2dc48c2e7ca572a5277e76f9d/composer.json#L23-L37

 * all `php-http/*` packages are related to sending events through HTTP - this is the bulk of the required dependencies
 * `jean85/pretty-package-versions` is used to retrieve installed package versions; we may drop it if we restrict to Composer 2 (and drop Composer 1 support - not sure it's desirable, but I would be in favor)
 * `psr/*` packages are just PHP-FIG interfaces, with no further dependencies
 * `symfony/options-resolver` is used to handle the config, and has no further dependencies apart from a deprecation handler (this may be droppable)
 * `symfony/polyfill-php80` is a simple polyfill, no sub-dependencies

Basically, we can decide to drop only two of them. reducing everything else by 3 tops. The core of the matter is the HTTP client, and that's a must. If you're having conflicts of any kind, we may investigate those, but I fear it's just out-of-date packages, and you can overcome that by installing older versions of the SDK.

**Comment by cleptric** [comment]
Hey @abhibeckert 

Over the last couple of months, we have already eliminated some dependencies.
By our [SDK Philosophy and Design Principles](https://develop.sentry.dev/sdk/philosophy/), we always strive to use as few dependencies as possible.
Could you point out some examples of our dependencies that caused you troubles?

We will re-evaluate our dependencies in an upcoming v4, but at the moment, I don't see much we could 
drop further.
